modules/oceanbase-ce/src/main/java/org/testcontainers/containers/OceanBaseContainer.java

```diff
@@ -0,0 +1,163 @@
+    private static final String DEFAULT_TEST_TENANT_NAME = "test";
+    private static final String DEFAULT_USERNAME = "root";
+    private static final String DEFAULT_PASSWORD = "";
+    private static final String DEFAULT_DATABASE_NAME = "test";
+
+    private boolean enableFastboot;
+    private String mode;
+    private String tenantName = DEFAULT_TEST_TENANT_NAME;
+    private String driverClassName = "com.mysql.cj.jdbc.Driver";
+
+    public OceanBaseContainer(String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    public OceanBaseContainer(DockerImageName dockerImageName) {
+        super(dockerImageName);
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+
+        addExposedPorts(SQL_PORT, RPC_PORT);
+    }
+
+    @Override
+    public Integer getMappedPort(int originalPort) {
+        return "host".equals(getNetworkMode()) ? originalPort : super.getMappedPort(originalPort);
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return driverClassName;
+    }
+
+    @Override
+    public String getJdbcUrl() {
+        return getJdbcUrl(DEFAULT_DATABASE_NAME);
+    }
+
+    public String getJdbcUrl(String databaseName) {
+        String additionalUrlParams = constructUrlParameters("?", "&");
+        String prefix = driverClassName.contains("mysql") ? "jdbc:mysql://" : "jdbc:oceanbase://";
+        return prefix + getHost() + ":" + getMappedPort(SQL_PORT) + "/" + databaseName + additionalUrlParams;
+    }
+
+    @Override
+    public String getDatabaseName() {
+        return DEFAULT_DATABASE_NAME;
+    }
+
+    @Override
+    public String getUsername() {
+        return DEFAULT_USERNAME + "@" + tenantName;
+    }
+
+    @Override
+    public String getPassword() {
+        return DEFAULT_PASSWORD;
+    }
+
+    @Override
+    protected String getTestQueryString() {
+        return "SELECT 1";
+    }
+
+    /**
+     * Enable fastboot.
+     *
+     * @return this
+     */
+    public OceanBaseContainer enableFastboot() {
+        this.enableFastboot = true;
+        return self();
+    }
+
+    /**
+     * Set the deployment mode, see <a href="https://hub.docker.com/r/oceanbase/oceanbase-ce">Docker Hub</a> for more details.
+     *
+     * @param mode the deployment mode
+     * @return this
+     */
+    public OceanBaseContainer withMode(String mode) {
+        this.mode = mode;
+        return self();
+    }
+
+    /**
+     * Set the non-system tenant to be created for testing.
+     *
+     * @param tenantName the name of tenant to be created
+     * @return this
+     */
+    public OceanBaseContainer withTenant(String tenantName) {
+        if (StringUtils.isEmpty(tenantName)) {
+            throw new IllegalArgumentException("Tenant name cannot be null or empty");
+        }
+        if (SYSTEM_TENANT_NAME.equals(tenantName)) {
+            throw new IllegalArgumentException("Tenant name cannot be " + SYSTEM_TENANT_NAME);
+        }
```

Comment:
wonder if those validations should be at the image level and avoid duplicating the validation.

Reply:
The response from the image may not be as direct as the error message here. It refers to the 'DEFAULT_DATABASE_NAME' in OracleContainer.